components/compositing/largest_contentful_paint.rs

```diff
@@ -0,0 +1,133 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::HashMap;
+
+use base::cross_process_instant::CrossProcessInstant;
+use compositing_traits::largest_contentful_paint_candidate::{
+    ImageCandidate, LCPCandidates, LargestContentfulPaint,
+};
+use webrender_api::{Epoch, PipelineId};
+
+#[derive(Default)]
+pub struct LargestContentfulPaintDetector {
+    lcp_calculators: HashMap<PipelineId, LargestContentfulPaintCalculator>,
```

Comment:
I would probably prefer `FnvHashMap` (it has a faster hasher compared to the default one, although afaik it is not necessarily the fastest one out there.
Is the LCP detector on the critical path and we should optimize this more later, or does performance not matter much here?